modin/pandas/groupby.py

```diff
@@ -314,30 +346,20 @@ def __getitem__(self, key):
             Column lookups on GroupBy with arbitrary Series in by is not yet supported.
         """
         kwargs = {**self._kwargs.copy(), "squeeze": self._squeeze}
-        # Most of time indexing DataFrameGroupBy results in another DataFrameGroupBy object unless circumstances are
-        # special in which case SeriesGroupBy has to be returned. Such circumstances are when key equals to a single
-        # column name and is not a list of column names or list of one column name.
-        make_dataframe = True
-        if self._drop and self._as_index:
-            if not isinstance(key, list):
-                key = [key]
-                kwargs["squeeze"] = True
+        # The rules of type deduction for the resulted object is the following:
+        #   1. If `key` is a list-like or `as_index is False`, then the resulted object is a DataFrameGroupBy
+        #   2. Otherwise, the resulted object is SeriesGroupBy
```

Comment:
Going from [pandas code](https://github.com/pandas-dev/pandas/blob/5f648bf1706dd75a9ca0d29f26eadfbb595fe52b/pandas/core/base.py#L223-L228)
<details><summary>... and parameters brute-forcing</summary>

```python
df = pandas.DataFrame({"a": [1, 1, 2, 2], "b": [3, 4, 5, 6], "c": [3, 2, 1, 0]})
# self._drop is False for the following 'by':
>>> df.groupby([1, 1, 2, 2])["b"]
<pandas.core.groupby.generic.SeriesGroupBy object at 0x0000017922EF9188>
>>> df.groupby([1, 1, 2, 2], as_index=False)["b"]
<pandas.core.groupby.generic.DataFrameGroupBy object at 0x0000017922EF9108>
>>> df.groupby([1, 1, 2, 2])[["b"]]
<pandas.core.groupby.generic.DataFrameGroupBy object at 0x0000017922EF9088>
# self._drop is True for the following 'by':
>>> df.groupby("a")["b"]
<pandas.core.groupby.generic.SeriesGroupBy object at 0x0000017922EF92C8>
>>> df.groupby("a", as_index=False)["b"]
<pandas.core.groupby.generic.DataFrameGroupBy object at 0x0000017922EF9288>
>>> df.groupby("a")[["b"]]
<pandas.core.groupby.generic.DataFrameGroupBy object at 0x0000017922EF9048>
```

</details>

Reply:
I like these examples, can we add them as comments in the comment here, but replacing pandas with Modin?

Reply:
I'm not sure about that, adding examples makes the comment look more complex than it actually is:
```python
# The rules of type deduction for the resulted object is the following:
#   1. If `key` is a list-like or `as_index is False`, then the resulted object is a DataFrameGroupBy
#   2. Otherwise, the resulted object is SeriesGroupBy
# Examples:
#   - drop: any, as_index: False, __getitem__(key: any) -> DataFrameGroupBy
#   - drop: any, as_index: any, __getitem__(key: list_like) -> DataFrameGroupBy
#   - drop: any, as_index: True, __getitem__(key: label) -> SeriesGroupBy
```
Just a list of rules looks pretty clean and all-covering. What do you think?

Reply:
That looks great @dchigarev !

Reply:
Would you be able to add the rules there? Thanks!

Reply:
added